grammar.js

```diff
@@ -889,7 +889,31 @@ module.exports = grammar({
         $._reserved_identifier,
       )),
       '!',
-      alias($.delim_token_tree, $.token_tree)
+      $.macro_invocation_contents,
+    ),
+
+    macro_invocation_contents: $ => choice(
+      $._macro_invocation_contents_parentheses,
+      $._macro_invocation_contents_square_brackets,
+      seq('{', repeat($._delim_tokens), '}'),
```

Comment:
It seemed atypical to pass "comma-separated expressions" to a `{...}`-style macro invocation (and I don't think I've encountered any use cases to want to parse those differently myself (yet)) so left it basically as-is